### DISCUSSION TITLE
Clearing unused fonts

### DISCUSSION THREAD

**Discussion by PavanLuca** [bleachbit/bleachbit#1645] [Ideas]
I like that it allows to delete unused languages so I was thinking that it could also allow deleting unused fonts.

**Comment by az0** [comment]
That's an interesting idea. The trick would be to identify which fonts are used, and that could be difficult. Many application uses fonts in different ways. For example, there may be one (important) LibreOffice document using a font, and if becomes impossible if it's on a removable drive.

Even just within LibreOffice, there are many kind of file formats that would need to be parsed to check: doc, docx, odt, ods, xls, pptx. This requires special code for each one, and it would be I/O and CPU resource intensive to read all the documents. To avoid re-scanning them would require caching, which takes up disk space.

Anyone have thoughts on better ways to find  which fonts are in use?

**Reply by PavanLuca** [reply]
If it's hard to code why not just give users the fonts list like the feature to remove unused languages? Of course you can write a warning telling that this feature is harmful if performed without a certain knowledge and the user must choose with caution which fonts to delete, deselecting both fonts in use from the system and the applications and fonts that will be used by external files. For now a manual configuration, another time an automated check. For safety I'd choose the manual configuration in any case.
